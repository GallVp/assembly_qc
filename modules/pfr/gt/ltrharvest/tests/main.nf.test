nextflow_process {

    name "Test Process GT_LTRHARVEST"
    script "../main.nf"
    process "GT_LTRHARVEST"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "gt"
    tag "gt/ltrharvest"

    test("homo_sapiens-genome_21_fasta-success") {

        setup {
            run("GT_SUFFIXERATOR") {
                script "../../../../pfr/gt/suffixerator"

                process {
                    """
                    input[0] = [
                        [ id:'test' ],
                        file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = GT_SUFFIXERATOR.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.tabout).match("tabout") },
                { assert process.out.gff3 == [] },
                { assert snapshot(process.out.fasta).match("fasta") },
                { assert snapshot(process.out.inner_fasta).match("inner_fasta") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

    test("stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.tabout != null },
                { assert process.out.gff3 == [] },
                { assert process.out.fasta != null },
                { assert process.out.inner_fasta != null },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

}
