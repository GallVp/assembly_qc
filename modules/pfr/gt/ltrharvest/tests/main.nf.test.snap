{
    "inner_fasta": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "inner.fasta:md5,9042f79dda17cca97ee04f30ed86c935"
                ]
            ]
        ],
        "timestamp": "2023-12-03T21:06:51.510422"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,8df72f2f0252e314180bde73e0bbae37"
            ]
        ],
        "timestamp": "2023-12-03T20:59:09.586626"
    },
    "tabout": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "test.tabout:md5,1dfc230e79b304c3766b63791cfa1122"
                ]
            ]
        ],
        "timestamp": "2023-12-03T21:06:51.39337"
    },
    "fasta": {
        "content": [
            [
                [
                    {
                        "id": "test"
                    },
                    "out.fasta:md5,c78256fe6d62aafec5d5ddedded1e60e"
                ]
            ]
        ],
        "timestamp": "2023-12-03T21:06:51.447436"
    }
}